scripts/bulk_update_leads_info.py

```diff
@@ -0,0 +1,105 @@
+parser.add_argument('--development', '-d', action='store_true',
+                    help='Use a development (testing) server rather than production.')
+parser.add_argument('--confirmed', '-c', action='store_true',
+                    help='Without this flag, the script will do a dry run without actually updating any data.')
+parser.add_argument('--disable-create', '-x', action='store_true',
+                    help='Prevent new lead creation. Update only exists leads.')
+args = parser.parse_args()
+
+log_format = "[%(asctime)s] %(levelname)s [%(name)s.%(funcName)s:%(lineno)d] %(message)s"
+if not args.confirmed:
+    log_format = 'DRY RUN: '+log_format
+logging.basicConfig(level=logging.INFO, format=log_format)
+logging.debug('parameters: %s' % vars(args))
+
+sniffer = csv.Sniffer()
+dialect = sniffer.sniff(args.csvfile.read(1024))
+args.csvfile.seek(0)
+c = csv.DictReader(args.csvfile, dialect=dialect)
+assert any(x in ('company', 'lead_id') for x in c.fieldnames), \
+    'ERROR: column "company" or "lead_id" is not found'
+
+
+api = CloseIO_API(args.api_key, development=args.development)
+
+try:
+    resp = api.get('custom_fields/lead/')
+except APIError as e:
+    logging.error('line: %d : %s' % (c.line_num, e))
+    raise
+
+available_custom_fieldnames = [x['name'] for x in resp['data']]
+new_custom_fieldnames = [x for x in [y.split('.')[1] for y in c.fieldnames if y.startswith('custom.')]
+                         if x not in available_custom_fieldnames]
+
+for field in new_custom_fieldnames:
+    try:
+        if args.confirmed:
+            api.post('custom_fields/lead/', data={'name': field, 'type': 'text'})
+        logging.info('added new custom field "%s"' % field)
+    except APIError as e:
+        logging.error('line: %d : %s' % (c.line_num, e))
+        raise
+
+for r in c:
+    payload = {'name': r['company'],
+               'url': r.get('url'),
+               'contacts': [{'name': r['contact%s_name' % x],
+                            'title': r['contact%s_title' % x],
+                            'phones': get_contact_info(x, r, 'phone', 'office'),
+                            'emails': get_contact_info(x, r, 'email', 'office'),
+                            'urls': get_contact_info(x, r, 'url', 'url')}
+                            for x in [y[7] for y in r.keys()
+                                      if re.match(r'contact[0-9]_name', y) and r[y]]],
+               'custom': {x.split('.')[1]: r[x] for x in [y for y in r.keys() if y.startswith('custom.')]}
+               }
+
+    lead = None
+    try:
+        if r.get('lead_id') is not None:
+            # exists lead
+            resp = api.get('lead/%s' % r['lead_id'], data={
+                'fields': 'id'
+            })
+            lead = resp['data']
+        else:
+            # first lead in the company
+            resp = api.get('lead', data={
+                'query': 'company: "%s" sort:created' % r['company'],
```

Comment:
this is fine, but to keep things consistent with how our UI generates queries, make it `company:"%s"`